middlewares/mirror.go

```diff
@@ -0,0 +1,149 @@
+package middlewares
+
+import (
+	"bytes"
+	"io/ioutil"
+	"net/http"
+	"sync/atomic"
+
+	"github.com/containous/traefik/log"
+	"github.com/containous/traefik/safe"
+	"github.com/containous/traefik/types"
+	"github.com/vulcand/oxy/forward"
+	"github.com/vulcand/oxy/utils"
+	"net/url"
+)
+
+//Mirror is a middleware that provides the request mirroring capabilities
+type Mirror struct {
+	BackendURL     *url.URL
```

Comment:
Can we make those fields also private? I think it's best practice to only export when it's really required.

Reply:
👍